app/ide-desktop/lib/dashboard/src/pages/authentication/SetUsername.tsx

```diff
@@ -26,13 +26,16 @@ export default function SetUsername() {
   const { email } = authProvider.usePartialUserSession()
   const backend = backendProvider.useRemoteBackendStrict()
   const { getText } = textProvider.useText()
+  const localBackend = backendProvider.useLocalBackend()
+  const supportsOffline = localBackend != null
 
   const [username, setUsername] = React.useState('')
 
   return (
     <AuthenticationPage
       data-testid="set-username-panel"
       title={getText('setYourUsername')}
+      supportsOffline={supportsOffline}
```

Comment:
might need docs for this prop - e.g. if it is always `localBackend != null`, is there a reason it needs to be on every page?

if not, what does it mean when this is accessible when the local backend is available?

Reply:
Technically, there is no reason 🙂, but I see AuthenticationPage more like a presentational component, so it's not his responsibility to know what `supportsOffline` means.

Ideally, we need something like a `<AppContext />` that stores: 

1. env vars
2. App level related parameters, like `supportLocalBackend`

Then we can consume this logic across the app and lower-level components will be unaware about this